compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -331,6 +331,8 @@ pub const BUILTIN_ATTRIBUTES: &[BuiltinAttribute] = &[
         optimize, AssumedUsed, template!(List: "size|speed"), optimize_attribute,
         experimental!(optimize),
     ),
+    // RFC 2867
+    gated!(instruction_set, AssumedUsed, template!(List: "arm::a32|arm::t32"), isa_attribute, experimental!(instruction_set)),
```

Comment:
@jonas-schievink so I have a feeling this is causing one of my issues with the E0775 error code test?

```
error[E0658]: the `#[instruction_set]` attribute is an experimental feature
 --> /checkout/obj/build/x86_64-unknown-linux-gnu/test/error-index.md:15478:1
  |
2 | #[instruction_set()] // error: expected one argument
  | ^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #74727 <https://github.com/rust-lang/rust/issues/74727> for more information
  = help: add `#![feature(isa_attribute)]` to the crate attributes to enable

error[E0776]: target does not support `#[instruction_set]`
 --> /checkout/obj/build/x86_64-unknown-linux-gnu/test/error-index.md:15478:1
  |
2 | #[instruction_set()] // error: expected one argument
  | ^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors
```

Reply:
I tried to do it with `Word` as the element instead of `List` previously but it didn't seem to work, might have been multiple issues working together though :thinking:

Reply:
You should use `#![feature(instruction_set)]` in the test to get rid of the first error.

The second error just looks like architecture support is checked before attribute syntax, which matches what your code is doing. I'm not sure how much validation is actually derived from the `template!` (the string is used in the error message, but might not actually be used to check attribute syntax).

Reply:
Oh then I made a mistake (I guess). The RFC says the feature gate is isa_attribute and the attribute is instruction_set.

Reply:
Ah, right, `#![feature(isa_attribute)]` should also fix this then. I think the RFC initially proposed `#[isa]` instead of `#[instruction_set]`.

Reply:
Yeah I remember reading some discussion that isa is easily confused for "is a"

Reply:
> The second error just looks like architecture support is checked before attribute syntax, which matches what your code is doing. I'm not sure how much validation is actually derived from the template! (the string is used in the error message, but might not actually be used to check attribute syntax). 

So any suggestions on how to get around this?

Reply:
Change the check so it does it the other way around?

Reply:
This codes outdated I changed it to `gated!(isa_attribute, AssumedUsed, template!(Word), isa_attribute, experimental!(instruction_set)),` but I still get the not supported on this architecture in the E0775 ui test

Reply:
Your code in typeck first checks whether the architecture supports `#[instruction_set]`, and only then check whether it is syntactically correct. There is no architecture check done in builtin_attrs.rs.

Reply:
So with the E0775.rs ui test I have `// only-armv4t-unknown-linux-gnueabi` at the top of the file. And I thought that meant I'd only get the compiler errors for that architecture. Because I wouldn't expect the error messages to be the same for an unsupported arch as a supported arch, and checking thumb_interworking first seems to be the quickest and most-general check to do first (arm v4/v5 isn't a lot of users). I could change it to do it the other way round and fix the errors that way if that's the better/easiest/preferred approach

Reply:
I think that would make sense. I'd usually expect syntax to be validated first, and independently of the target, and then have semantic checks performed.